src/GitVersionCore/VersionCalculation/NextVersionCalculator.cs

```diff
@@ -77,6 +78,12 @@ public SemanticVersion FindVersion(GitVersionContext context)
                 }
             }
 
+            if (context.IsLocalBuild && taggedSemanticVersion == null && context.Repository.RetrieveStatus().IsDirty)
```

Comment:
This is why this is superior to our current solution we have with our own msbuild target.
We can check if the repo is dirty in GitVersion 😍 

cc @zionyx

Reply:
Though I wonder if that's what we want.
We properly want to have all local build marked with `local.timestamp` when I think about our use case for local nuget repository source.